Improve Ubuntu instructions (#40)
uyuni-project/uyuni-docs: modules/client-configuration/pages/clients-ubuntu.adoc

This section contains information about registering Salt clients running {ubuntu} operating systems.

{productname} supports {ubuntu} 16.04 LTS and 18.04 LTS Clients using Salt.
Traditional clients are not supported.

[IMPORTANT]
====
Canonical does not endorse or support {productname}.
====

Bootstrapping is supported for starting {ubuntu} clients and performing initial state runs such as setting repositories and performing profile updates.
However, the root user on {ubuntu} is disabled by default, so to use bootstrapping, you will require an existing user with [command]``sudo`` privileges for Python.

== Prepare to Register

Some preparation is required before you can register {ubuntu} clients to the {productname} Server.

ifeval::[{suma-content} == true]

> .Procedure: Adding client tools channels

Reviewer: Let's split this into two, one for the webui, and one for the CLI. Then we can move L28 up to before both of them.

Reviewer 2: ditto

In the {productname} {webui}, navigate to menu:Software[Channel List > All].
Verify that you have a base channel and a child channel for your architecture.

For example, for {ubuntu} 18.04:

* Base channel: [systemitem]``ubuntu-18.04-pool for amd64``
* Child channel: [systemitem]``Ubuntu-18.04-SUSE-Manager-Tools for amd64``

.Procedure: Creating Custom Channels and Repositories

You need to manually create three repositories:

* For main
* For main updates
* For main security

> The following examples are for {ubuntu} 18.04 (bionic). Make sure you change the values to match the {ubuntu} version you want to use, and make sure you replace ```ubuntumirror.example.com``` with a valid mirror.

Reviewer: ```suggestion
These examples are for {ubuntu} 18.04 (bionic). 
Make sure you change the values to match the {ubuntu} version you want to use.
You will also need to replace ```ubuntumirror.example.com``` with a valid mirror.
```

ifeval::[{suma-content} == true]
.Procedure: Alternative Method for Adding {ubuntu} Channels and Repositories (Unsupported)

> .  

Author: This is a hack until we find why how to avoid the title procedure getting inside the warning.

Reviewer: We should avoid admonitions in procedures. Can't we just put it above the procedure title?

Reviewer 2: https://github.com/SUSE/spacewalk/issues/10522

[WARNING]
====
The ``spacewalk-utils`` package is not supported by {suse}, including the ``spacewalk-common-channels`` tool.
You will not be supported if your system becomes inoperable by using these tools.
====

. At the command prompt on the {productname} Server, as root, install the [systemitem]``spacewalk-utils`` package:
+
----
zypper in spacewalk-utils
----
. Add the {ubuntu} channels.

For {ubuntu} 16.04:

----
spacewalk-common-channels \
ubuntu-1604-amd64-main \
ubuntu-1604-amd64-main-updates \
ubuntu-1604-amd64-main-security
----

For {ubuntu} 18.04:

----
spacewalk-common-channels \
ubuntu-1804-amd64-main \
ubuntu-1804-amd64-main-updates \
ubuntu-1804-amd64-main-security
----

> When you have the channels set up, associate each channel with the appropriate repository, and synchronize then channels (optionally configure scheduled syncronization).

Reviewer: ```suggestion
When you have the channels set up, associate each channel with the appropriate repository, and synchronize them.
You can also configure a synchronization schedule.
```

> To do this, proceed to menu:Software[Manage > Channels], and for each channel you created, click on it and:

Reviewer: ```suggestion
To manually synchronize the channels, navigate to menu:Software[Manage > Channels].
Click on each channel in the list, and:
```

Reviewer 2: Just "Click" (instead of "Click on").

> . Navigate to the [guimenu]``Repositories``

Reviewer: ```suggestion
. Navigate to the [guimenu]``Repositories`` tab.
```

> . Navigate to the [guimenu]``Sync``and click btn:[Sync Now] to begin synchronization immediately.

Reviewer: ```suggestion
. Navigate to the [guimenu]``Sync`` sub-tab.
. Click btn:[Sync Now] to begin synchronization immediately.
```

Reviewer 2: I think we usually write "subtab" (w/o a dash).

> . You can also setup a scheduled syncronization from this screen.

Reviewer: ```suggestion
. You can also create a synchronization schedule from this screen.
```

Synchronization can sometimes take several hours.
====

When you have the channels set up, associate each channel with the appropriate repository, and synchronize then channels (optionally configure scheduled syncronization).

To do this, proceed to menu:Software[Manage > Channels], and for each channel you created, click on it and:

. Navigate to the [guimenu]``Repositories``
. Navigate to the [guimenu]``Sync``and click btn:[Sync Now] to begin synchronization immediately.
. You can also setup a scheduled syncronization from this screen.

> 

Reviewer: See earlier edits (why is this repeated, though?)

Reviewer 2: ditto

> == Monitor syncronization progress

Reviewer: ```suggestion
== Monitor Synchronization Progress
```

Author: https://github.com/SUSE/spacewalk/issues/10239#issuecomment-568176950
